sound/soc/sof/core.c

```diff
@@ -528,6 +542,7 @@ int snd_sof_device_remove(struct device *dev)
 	 * removed only after the topology has been unloaded.
 	 */
 	snd_sof_remove(sdev);
+#endif
```

Comment:
No, this cannot work.
we don't want to have any dependencies on SOF_HDA in the core
And we would have a different behavior for machine drivers like sof-rt5682 depending on whether the SOF_HDA option is enabled.